src/utils/fetchClient.ts

```diff
@@ -29,7 +29,13 @@ function request<T>(
   // for a demo purpose we emulate a delay to see if Loaders work
   return wait(300)
     .then(() => fetch(BASE_URL + url, options))
-    .then(response => response.json());
+    .then(response => {
+      if (!response.ok) {
+        throw new Error('Something wrong');
```

Comment:
The error message 'Something wrong' is too generic. Consider providing more specific error information, such as the status code or response text, to help with debugging and user feedback.